### DISCUSSION TITLE
Questions about laser settings

### DISCUSSION THREAD

**Discussion by chengzi0o0** [BLAST-WarpX/warpx#5511] [Q&A] [answered]
Hi everyone!

I’m new to laser-plasma science and using WarpX, and I’ve run into some confusion while working with it. I would really appreciate some guidance.

I noticed that the plane of the antenna emits lasers in both the front and back directions, and there are localized sources on the plane. In most cases, only one side of the laser is used. In my personal understanding, the antenna should be placed as far away from the particle as possible to avoid significant influence on the particle. For example, when the laser pulse width is long enough, the particle may pass through the plane (plasma expansion) and be affected by the laser in the opposite direction, which is not the expected situation. However, I saw that the antenna in the examples [*inputs_test_2d_laser_ion_acc*](https://github.com/ECP-WarpX/WarpX/blob/development/Examples/Physics_applications/laser_ion/inputs_test_2d_laser_ion_acc) and [*inputs_base_3d*](https://github.com/ECP-WarpX/WarpX/blob/development/Examples/Physics_applications/laser_acceleration/inputs_base_3d) are set in the pre-plasma and electrons respectively, which makes me a little confused. Maybe the laser setting has less influence in these examples?

In addition, I am now considering using an oblique incident laser to hit a planar target, but the antenna plane seems to be truncated by the boundary of the simulation area. I need to set the inclined laser plane carefully, and I am also worried about some effects in the area behind the antenna plane mentioned above(In addition to plasma expansion, the reflected light after the laser hits the target will overlap with the laser on the other side of the plane.). Although it may not have much impact on the area I am interested in, this part of the area will become redundant. Is there a more suitable method for setting up oblique incident lasers?

I’m still learning, so any help would be really appreciated. Thank you in advance!

**Comment by ax3l** [comment, marked as answer]
Hi @chengzi0o0,

Thank you for your question and welcome to WarpX and laser-plasma science! :tada: 

Yes, all you are saying is correct. The way we initialize laser pulses with an antenna (virtual current) emits in both directions from the laser initialization plane. The side not needed can simply be absorbed in field boundaries, e.g., PML.

You are correct, you want to avoid that relevant plasma particle dynamics is interacting with the initialization plane while it is active. Also, the fields we initialize are derived usually for vacuum. For many scenarios, as the cases you showed above, we might after a certain time or for a certain region not be interested in particles that move against the laser direction, either because we are looking at processes in forward direction, know the particles will not reflect back, or other physical reason to neglect the physics in some parts of the box because it will not influence the rest. In the cases you showed, the plasma density is very low and not necessarily significant.

But, the examples you link are also at very coarse resolution and size to fit in cases we can demonstrate quickly and run often as tests. Yes, in a concrete case definitely check and potentially move the plane further away if it influences relevant physical observables you are studying. Essentially, you can save computational time and cut closer if you see it has no influence on your interaction region, otherwise you need larger boxes (more cells) and more computational resources.

> In addition, I am now considering using an oblique incident laser to hit a planar target, but the antenna plane seems to be truncated by the boundary of the simulation area.

Correct, you will need a wider transverse simulation box for oblique laser incidents, so you do not crop the transverse profile of your incoming laser pulse.

> I am also worried about some effects in the area behind the antenna plane mentioned above [...]

Using absorbing boundary conditions like PMLs should well absorb the field also here, even if it is not perfectly 90 degrees to the boundaries.

> In addition to plasma expansion, the reflected light after the laser hits the target will overlap with the laser on the other side of the plane.

Then you need to make the box wider and move the plane further away from the target.

> Although it may not have much impact on the area I am interested in, this part of the area will become redundant.

Sounds like you like to try it out: get closer and further and check the physics you like to study is truly not affected. You can do a lot of tricks and shortcuts to save compute resources in modeling - as long as you can proof it truly keeps what you study in good shape ;)

**Reply by chengzi0o0** [reply]
Your explanation was clear and really cleared up my confusion about the laser settings. Thank you so much for your detailed response!
